src/utils/decorators.ts

```diff
@@ -48,6 +48,11 @@ export function createClassPropertyDecorator(
 					}
 				}
 			};
+			if (arguments.length < 3) {
+				// Typescript target is ES3, so it won't define property for us
+				Object.defineProperty(target, key, descriptor);
```

Comment:
Hey @andykog, what will this fix? If target is ES3, then `Object.defineProperty` should not be available in the first place right? I guess people should just not compile for ES3. So maybe we should just throw an exception here?